barretenberg/cpp/src/barretenberg/dsl/acir_format/acir_to_constraint_buf.cpp

```diff
@@ -82,13 +123,60 @@ T deserialize_any_format(std::vector<uint8_t>&& buf,
     return decode_bincode(std::move(buf));
 }
 
-/**
- * @brief Deserializes a `Program` from bytes, trying `msgpack` or `bincode` formats.
- * @note Ignores the Brillig parts of the bytecode when using `msgpack`.
- */
-Acir::Program deserialize_program(std::vector<uint8_t>&& buf)
+AcirFormat circuit_serde_to_acir_format(Acir::Circuit const& circuit)
+{
+    AcirFormat af;
+    // `varnum` is the true number of variables, thus we add one to the index which starts at zero
+    af.varnum = circuit.current_witness_index + 1;
```

Comment:
Ok I vaguely recall this now. varnum is the number of witnesses we thought we'd have before some acir optimizations? Do we need varnum for any reason?